### PR TITLE
Use global compiler optimization flags instead of defining them for every plugin/binary

### PR DESCRIPTION
**Pull request by nE0sIghT**
In Gentoo Linux every user defines compiler flags that best suits his system.
It's usually set to recommended "-O2 -match=native", but some can try another optimizations (or even other compilers with other flags).

Currently, pcsx2 build system force "-O2" flag for most plugins. That patch allow to disable all extra CFLAGS by providing DISABLE_OPTIMIZATION cmake option.

### CODE REVIEW

**Comment by gregory38** [comment]
I don't like current patch. I would prefer to make a standard optimization variable (that contains 02). This way you just need to disable a single variable.

```
In Gentoo Linux every user defines compiler flags that best suits his system.
```
Sure but each time someone change a flag/compiler, it just finishes with more crashes (for example fomit-frame-pointer...). So I'm not really sure. By the way there is already a way to provide extra flags, not sure it worths it to remove O2 stuff.

**Comment by nE0sIghT** [comment]
I pushed commit with new OPTIMIZATION_FLAG variable (default to -O2).

> Sure but each time someone change a flag/compiler, it just finishes with more crashes

For that case we have filters in Gentoo build system to strip unwanted flags. For example, [-mnoavx was added for firefox and thunderbird](https://bugs.gentoo.org/show_bug.cgi?id=382023) some time ago. But such things is normal for source based distr :-)

File: cmake/BuildParameters.cmake [hunk 1]
```diff
@@ -124,14 +124,13 @@ if(${PCSX2_TARGET_ARCHITECTURES} MATCHES "i386")
     #     - Only plugins. No package will link to them.
     set(CMAKE_POSITION_INDEPENDENT_CODE OFF)
 
-    if (DISABLE_ADVANCE_SIMD)
-        set(ARCH_FLAG "-msse -msse2 -march=i686")
-    else()
+    if (NOT DISABLE_ADVANCE_SIMD)
```
Comment (gregory38) [inline cmake/BuildParameters.cmake]: Why did you remove the ARCH_FLAG variable.

I don't think it will compile (run) anymore in DISABLE_ADVANCE_SIMD mode.

File: cmake/BuildParameters.cmake [hunk 2]
```diff
@@ -143,9 +142,7 @@ elseif(${PCSX2_TARGET_ARCHITECTURES} MATCHES "x86_64")
     # SuperVU will not be ported
     set(DISABLE_SVU TRUE)
 
-    if (DISABLE_ADVANCE_SIMD)
-        set(ARCH_FLAG "-msse -msse2")
-    else()
+    if (NOT DISABLE_ADVANCE_SIMD)
```
Comment (gregory38) [inline cmake/BuildParameters.cmake]: Same comment as above

File: cmake/BuildParameters.cmake (hunk 1 above)
Comment (nE0sIghT) [inline cmake/BuildParameters.cmake]: Just rechecked - compiles fine for me on amd64 host using build.sh:
```
./build.sh --release --cross-multilib --no-simd --clean
Doing a clean build.
Building pcsx2 with -DCMAKE_BUILD_PO=FALSE -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=cmake/linux-compiler-i386-multilib.cmake -DDISABLE_ADVANCE_SIMD=TRUE

```
As i see "cmake/linux-compiler-i386-multilib.cmake" do all needed stuff and -march is optimization flag

Comment (gregory38) [inline cmake/BuildParameters.cmake]: I don't see anything on the arch flag inside cmake/linux-compiler-i386-multilib.cmake
The typical usecases are 
1/ build on a machine that support avx for users with older CPU
2/ Disable AVX/SSE for valgrind or virtual box

SSE2 and March=i686 is the minimal acceptable optimization. It doesn't hurt gentoo as all peoples will use the march=native path anyway.

In my opinion, your build isn't fully portable.

Comment (nE0sIghT) [inline cmake/BuildParameters.cmake]: I see now what i break with that change in generic case.
Pushed new commit - could we wrap those block with "if(NOT DEFINED ARCH_FLAG)" to allow overriding hardcoded values?
